dartagnan/src/test/java/com/dat3m/dartagnan/c/C11OrigLocksTest.java

```diff
@@ -0,0 +1,33 @@
+package com.dat3m.dartagnan.c;
+
+import com.dat3m.dartagnan.configuration.Arch;
+import com.dat3m.dartagnan.configuration.OptionNames;
+import com.dat3m.dartagnan.utils.Result;
+import com.dat3m.dartagnan.utils.rules.Provider;
+import com.dat3m.dartagnan.utils.rules.Providers;
+import com.dat3m.dartagnan.wmm.Wmm;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.sosy_lab.common.configuration.Configuration;
+import org.sosy_lab.common.configuration.InvalidConfigurationException;
+
+@RunWith(Parameterized.class)
+public class C11OrigLocksTest extends C11LocksTest {
+
+    public C11OrigLocksTest(String name, Arch target, Result expected) {
+        super(name, target, expected);
+    }
+
+    @Override
+    protected Configuration getConfiguration() throws InvalidConfigurationException {
+        return Configuration.builder()
+                .copyFrom(super.getConfiguration())
+                .setOption(OptionNames.INIT_DYNAMIC_ALLOCATIONS, "true")
+                .build();
+    }
```

Comment:
You said there are 3 configurations that makes sense. We are missing one where `OptionNames.INIT_DYNAMIC_ALLOCATIONS` is set to false (same for the LFDS test)

Reply:
Well, I said there are (at least) 3 configurations but I didn't intend to have all of them. Technically, I could come up with even more combinations for all tests under all memory models :P. I think at some point you have to draw the line.
Anyway, I can add this one extra case for C11.

Reply:
I thought we were missing the variant that was actually enabling the new code. Now I understand we have "new code + new model" and "old code + old model" which are the two most important cases, so feel free to ignore this review comment